pyteal/ast/abi/util.py

```diff
@@ -205,3 +205,30 @@ def type_spec_from_annotation(annotation: Any) -> TypeSpec:
         return TupleTypeSpec(*(type_spec_from_annotation(arg) for arg in args))
 
     raise TypeError("Unknown annotation origin: {}".format(origin))
+
+
+T = TypeVar("T", bound=BaseType)
+
+
+def make(t: type[T]) -> T:
+    """Create a new instance of an ABI type. The type to create is determined by the input argument,
+    which must be a fully-specified annotation of type's class.
+
+    For example:
+        .. code-block:: python
+
+                # both of these are equivalent
+                myTuple = abi.make(abi.Tuple2[abi.Uint64, abi.DynamicArray[abi.Bool]])
+                myTuple = abi.Tuple(abi.TupleTypeSpec(abi.Uint64TypeSpec(), abi.DynamicArrayTypeSpec(abi.BoolTypeSpec())))
+
+    This is purely a convenience method over instantiating the type directly, which can be cumbersome
+    due to the lengthy TypeSpec class names.
+
+    Args:
+        t: A fully-specified annotation of a subclass of abi.BaseType, meaning all generic parameters
+            must be given values.
+
+    Returns:
+        A new instance of the given type class.
```

Comment:
While this is technically correct, the word `given` to me sounds like the input (and we're given an annotation not a type class). So maybe slightly clearer would be:

```
    Returns:
        A new instance of the corresponding type class.
```

Reply:
I opted to drop the use of annotation elsewhere, since an annotation is a class and both terms added confusion.